### ISSUE TITLE
app.If() is taken with care as it always evaluates the element list.

### ISSUE THREAD

**Issue by oderwat** [maxence-charriere/go-app#632]
It is somewhat obvious but can lead to strange bugs anyway. If one uses `app.If()`, it always evaluates the element list. This may decrease performance and even lead to crashes. So it may be better if we have an implementation that takes functions instead.

**Comment by maxence-charriere** [comment]
I feel If is already heavy to use in terms of syntax. Changing the API to take a func would worse the thing.

**Comment by oderwat** [comment]
Well, I have such implementation for evaluation in my version and it works well. I was thinking about it being too cumbersome but having the render routine in one long streak makes it easier to understand what happens. I am undecided and this is also why I didn't make a PR for it. I almost think the current offerings of If() / elseIf() / else() is an error because the evaluations are always overhead.

**Comment by oderwat** [comment]
I think that using something like this makes more sense than the `app.If()` construct and that `app.If()` should be deprecated and give such an example ;-)

```go
func (c *thing) Render() app.UI {
	return func(e bool) app.UI {
		if e {
			return app.Div()."something is true")
		} else {
			return nil
		}
	}(something)
}    
```

**Comment by ZackaryWelch** [comment]
I agree with this, but I don't know if it's fixable. It has lead to me avoid using app.lf, or at least being very mindful when I do.

**Comment by justinfx** [comment]
fwiw I have already hit crashes 2 times in my first app, by using `app.If()` and not remember that its always evaluated, and that code path has a null pointer when its not the active branch. I've switched to doing my own branch ahead of time. It would be better to deprecate `app.If` and provide an alternate `app.IfFunc` to take an anonymous function and defer the evaluation until its the active branch.

**Comment by suntong** [comment]
Let's just start with changing the Hello World example from README.md:

https://github.com/maxence-charriere/go-app#declarative-syntax

> Here is a Hello World component that takes an input and displays its value in its title

Else, this will be the first thing people would try and then _form the habit of using it_, thinking this is the recommended approach.
